lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const CSSMatchedStyles = require('../../../lib/web-inspector').CSSMatchedStyles;
+const Gatherer = require('../gatherer');
+const FONT_SIZE_PROPERTY_NAME = 'font-size';
+
+/**
+ * @param {!Node} node Top document node
+ * @returns {Node}
+ */
+function findBody(node) {
+  const queue = [node];
+
+  while (queue.length > 0) {
+    const currentNode = queue.shift();
+
+    if (currentNode.nodeName === 'BODY') {
+      return currentNode;
+    }
+
+    if (currentNode.children) {
+      currentNode.children.forEach(child => queue.push(child));
+    }
+  }
+
+  return null;
+}
+
+/**
+ * @param {!Node} node
+ * @returns {!Array<!Node>}
+ */
+function flattenTree(node) {
+  const flat = [node];
+
+  for (let i = 0; i < flat.length; i++) {
+    const currentNode = flat[i];
+
+    if (currentNode.children) {
+      currentNode.children.forEach(child => {
+        child.parentNode = currentNode;
+        flat.push(child);
+      });
+    }
+  }
+
+  return flat;
+}
+
+/**
+ * Get list of all nodes from the document body.
+ *
+ * @param {!Object} driver
+ * @returns {!Array<!Node>}
+ */
+function getAllNodesFromBody(driver) {
```

Comment:
do you think you could modify https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/gather/driver.js#L739-L745 for your use case or is it much easier to work directly with the protocol as you're doing here?

it'd be nice to avoid collecting to many ad-hoc methods of retrieving DOM elements

Reply:
bump for thoughts on this :)

Reply:
`getElementsInDocument` filters text nodes out and puts everything into `Element`s (throwing away a lot of metadata returned by the `DOM. getFlattenedDocument`) so it was really hard for me to reuse it. Instead, I have extracted the `this.sendCommand('DOM.getFlattenedDocument', {depth: -1, pierce}) ` part as a new  driver method (`getNodesInDocument`) that `getElementsInDocument` depends on.

Reply:
Ah ok, that's cool :)